tardis/energy_input/tests/test_gamma_ray_channel.py

```diff
@@ -0,0 +1,235 @@
+    )
+
+    npt.assert_allclose(shell_masses[0], desired)
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56", "Fe-52", "Cr-48"])
+def test_isotope_dicts(gamma_ray_model_state, nuclide_name):
+    """
+    Function to test if the right names for the isotopes are present as dictionary keys.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+    raw_isotope_abundance, cell_masses = gamma_ray_model_state
+    isotope_dict = create_isotope_dicts(raw_isotope_abundance, cell_masses)
+
+    for isotope_dict in isotope_dict.values():
+        assert nuclide_name.replace("-", "") in isotope_dict.keys()
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56", "Fe-52", "Cr-48"])
+def test_inventories_dict(gamma_ray_model_state, nuclide_name):
+    """
+    Function to test if the inventories dictionary is created correctly.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    nuclide_name: Name of the nuclide.
+    """
+
+    nuclide = rd.Nuclide(nuclide_name)
+    raw_isotope_abundance, cell_masses = gamma_ray_model_state
+    isotope_dict = create_isotope_dicts(raw_isotope_abundance, cell_masses)
+    inventories_dict = create_inventories_dict(isotope_dict)
+
+    Z, A = nuclide.Z, nuclide.A
+    raw_isotope_abundance_mass = raw_isotope_abundance.apply(
+        lambda x: x * cell_masses, axis=1
+    )
+
+    mass = raw_isotope_abundance_mass.loc[Z, A][0]
+    isotope_inventory = rd.Inventory({nuclide.nuclide: mass}, "g")
+
+    if nuclide_name in inventories_dict[0].contents:
+        assert (
+            inventories_dict[0].contents[nuclide_name]
+            == isotope_inventory.contents[nuclide_name]
+        )
+
+
+@pytest.mark.parametrize("nuclide_name", ["Ni-56"])
+def test_mass_energy_conservation(
+    gamma_ray_model_state, atomic_dataset, nuclide_name
+):
+    """
+    Function to test if the mass-energy conservation is satisfied.
+    Parameters
+    ----------
+    simulation_setup: A simulation setup which returns a model.
+    atomic_dataset: Tardis atomic-nuclear dataset
+    nuclide_name: Name of the nuclide."""
+
+    raw_isotope_abundance, cell_masses = gamma_ray_model_state
+    gamma_ray_lines = atomic_dataset.decay_radiation_data
+    isotope_dict = create_isotope_dicts(raw_isotope_abundance, cell_masses)
+    inventories_dict = create_inventories_dict(isotope_dict)
+    total_decays = calculate_total_decays(inventories_dict, 1 * u.d)
+    isotope_decay_df = create_isotope_decay_df(total_decays, gamma_ray_lines)
+
+    grouped_isotope_df = isotope_decay_df.groupby(
+        level=["shell_number", "isotope"]
+    )
+
+    parent_isotope = (
+        grouped_isotope_df.get_group((0, nuclide_name.replace("-", "")))[
+            "energy_per_channel_keV"
+        ].sum()
+        * (u.keV).to(u.MeV)
+        * u.MeV
+    )
+
+    neutrino_energy = 0.41 * u.MeV
```

Comment:
To test other isotopes, this should be parametrized

Reply:
I should check if there is a paper giving the values for other decays like Fe52 etc. But for now, I have added only test for one isotope.